components/payments/cmd/connectors/internal/connectors/bankingcircle/task_fetch_payments.go

```diff
@@ -73,11 +77,17 @@ func ingestBatch(
 
 		paymentType := matchPaymentType(paymentEl.Classification)
 
+		precision, ok := supportedCurrenciesWithDecimal[paymentEl.Transfer.Amount.Currency]
+		if !ok {
+			logger.Errorf("currency %s is not supported", paymentEl.Transfer.Amount.Currency)
+			metricsRegistry.ConnectorCurrencyNotSupported().Add(ctx, 1, metric.WithAttributes(connectorAttrs...))
+			continue
+		}
+
 		var amount big.Float
 		amount.SetFloat64(paymentEl.Transfer.Amount.Amount)
-
 		var amountInt big.Int
-		amount.Mul(&amount, big.NewFloat(100)).Int(&amountInt)
+		amount.Mul(&amount, big.NewFloat(math.Pow(10, float64(precision)))).Int(&amountInt)
```

Comment:
Complicated stuff.
Why not 
amountInt := amount.Mul(&amount, big.NewFloat(math.Pow(10, float64(precision))))

Reply:
Sorry, the float...